src/lib/http-agent.js

```diff
@@ -28,23 +28,21 @@ const DEFAULT_HTTPS_PORT = 443
 // 50 seconds
 const AGENT_PORT_TIMEOUT = 50
 
-const getAgent = async ({ httpProxy, certificateFile, exit }) => {
+const tryGetAgent = async ({ httpProxy, certificateFile }) => {
   if (!httpProxy) {
     return
   }
 
   let proxyUrl
   try {
     proxyUrl = new URL(httpProxy)
-  } catch (error) {
-    log(NETLIFYDEVERR, `${httpProxy} is not a valid URL`)
-    exit(1)
+  } catch {
+    return { error: `${httpProxy} is not a valid URL` }
```

Comment:
The `error` instance might be useful here for debugging, i.e. we might want to return it?

```js
  } catch (error) {
    return { error: `${httpProxy} is not a valid URL`, message: error.message }
```

Reply:
I'm not sure about returning the entire error object in response considering that most errors occurring in this function are known before-hand.

Reply:
This would be returning only the error message string, not the entire error object.
The purpose would be to show why the `httpProxy` is not a valid URL: is the scheme missing? is there some other syntax error? 
However, this is not a blocker, so please feel free to ignore :+1:

Reply:
Oh! I misunderstood. Returning the error message definitely makes sense.

I think I got confused since a few places in this file already had the response with the error message and thought that you wanted to add the stack too.

Reply:
Will add it in another PR soon since this one is merged.